tests/main.test.ts

```diff
@@ -16,16 +16,33 @@ beforeEach(async () => {
   process.env.COREPACK_DEFAULT_TO_LATEST = `0`;
 });
 
-it(`should refuse to download a package manager if the hash doesn't match`, async () => {
-  await xfs.mktempPromise(async cwd => {
-    await xfs.writeJsonPromise(ppath.join(cwd, `package.json` as Filename), {
-      packageManager: `yarn@1.22.4+sha1.deadbeef`,
+describe(`should refuse to download a package manager if the hash doesn't match`, () => {
+  it(`the one defined in "devEngines.packageManager" field`, async () => {
+    await xfs.mktempPromise(async cwd => {
+      await xfs.writeJsonPromise(ppath.join(cwd, `package.json` as Filename), {
+        devEngines: {
+          packageManager: {name: `yarn`, version: `1.22.4+sha1.deadbeef`},
```

Comment:
Does adding a hash to the `version` value break other tools that support `devEngines`? For example, when pnpm 10.2.1 reads a `devEngines` with a value of:

```json
"devEngines": {
  "packageManager": {
    "name": "pnpm",
    "version": "10.2.1+sha1.48adf39a4ab751eda7b73b99447d1f0b6d227e02"
  }
}
```

does it still pass pnpm’s validation that the correct package manager version is specified?

Reply:
According to semver spec, it shouldn’t, but anyway that’s off topic for this PR, if it’s not compatible with the package manager they’re using, users won’t supply it